src/core/struct_tree.js

```diff
@@ -196,18 +212,35 @@ class StructTreePage {
       return;
     }
     const id = this.pageDict.get("StructParents");
-    if (!Number.isInteger(id)) {
+    const ids = this.root.structParentIds?.get(pageRef);
```

Comment:
Similar to the comment above, you need to handle `pageRef` being `null` here:
```suggestion
    const ids =
      pageRef instanceof Ref && this.root.structParentIds?.get(pageRef);
```